vms/proposervm/block/block.go

```diff
@@ -4,29 +4,36 @@
 package block
 
 import (
+	"bytes"
 	"errors"
 	"fmt"
 	"time"
 
 	"github.com/ava-labs/avalanchego/ids"
 	"github.com/ava-labs/avalanchego/staking"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
 	"github.com/ava-labs/avalanchego/utils/hashing"
 	"github.com/ava-labs/avalanchego/utils/wrappers"
 )
 
 var (
 	_ SignedBlock = (*statelessBlock)(nil)
 
-	errUnexpectedSignature = errors.New("signature provided when none was expected")
-	errInvalidCertificate  = errors.New("invalid certificate")
+	errUnexpectedSignature        = errors.New("signature provided when none was expected")
+	errInvalidCertificate         = errors.New("invalid certificate")
+	errInvalidBlockEncodingLength = errors.New("block encoding length must be greater than zero bytes long")
+	errFailedToParseVRFSignature  = errors.New("failed to parse VRF signature")
 )
 
 type Block interface {
 	ID() ids.ID
 	ParentID() ids.ID
 	Block() []byte
 	Bytes() []byte
+	// VerifySignature validates the correctness of the VRF signature.
+	VerifySignature(*bls.PublicKey, []byte, ids.ID, uint32) bool
```

Comment:
Does it make sense to actually add this to the block interface? Or would it make more sense to just use the VRF signature on the `SignedBlock` interface and perform the verification then.

Reply:
given that it's being used only on signed blocks, I'll move it to the `SignedBlock` interface. thanks!